src/Symfony/Component/Form/FormFactory.php

```diff
@@ -126,7 +126,7 @@ public function createBuilderForProperty($class, $property, $data = null, array
 
         // user options may override guessed options
         if ($typeGuess) {
-            $options = array_merge($typeGuess->getOptions(), $options);
+            $options = array_replace_recursive($typeGuess->getOptions(), $options);
```

Comment:
> But this may not be the case for all other options where the usage of some keys may be mutually exclusive.

Sorry :/ I don't see what I missing here, would you show me an example, please? I've done a simple test (https://3v4l.org/5FvnM) but I don't see that is wrong with `array_replace_recursive()` function :/